File: src/main/java/guru/xgm/language/lni/lang/folding/LniCodeFoldingSettings.java
package guru.xgm.language.lni.lang.folding;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.SettingsCategory;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import lombok.Getter;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

@Setter
@Getter
@State(name = "LniCodeFoldingSettings", storages = @Storage("editor.xml"), category = SettingsCategory.CODE)
public class LniCodeFoldingSettings implements PersistentStateComponent<LniCodeFoldingSettings> {
    private boolean foldItems = false;

    public static LniCodeFoldingSettings getInstance() {
        return ApplicationManager.getApplication().getService(LniCodeFoldingSettings.class);
    }

    @Override
    public LniCodeFoldingSettings getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull final LniCodeFoldingSettings state) {
        XmlSerializerUtil.copyBean(state, this);
    }

}

File: src/main/kotlin/raft/war/language/lni/lang/folding/LniCodeFoldingSettings.kt
package raft.war.language.lni.lang.folding

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.SettingsCategory
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "LniCodeFoldingSettings", storages = [Storage("editor.xml")], category = SettingsCategory.CODE)
class LniCodeFoldingSettings : PersistentStateComponent<LniCodeFoldingSettings> {
    var isFoldItem: Boolean = false
    var isFoldList: Boolean = false

    override fun getState(): LniCodeFoldingSettings {
        return this
    }

    override fun loadState(state: LniCodeFoldingSettings) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        val instance: LniCodeFoldingSettings
            get() = ApplicationManager.getApplication().getService(LniCodeFoldingSettings::class.java)
    }
}
